src/test_linked_list.py

```diff
@@ -2,55 +2,64 @@
 
 
 test_check_pop_params_table = [
+    """Checks the pop function is returning expected results"""
```

Comment:
You should definitely not have documentation strings in your params tables, and you shouldn't call your params tables "test_" anything. Pytest is expecting anything called "test_" to be callable.